src/Doctrine/EntityDetails.php

```diff
@@ -21,14 +21,9 @@
  */
 final class EntityDetails
 {
-    private $metadata;
-
-    /**
-     * @param ClassMetadata|LegacyClassMetadata $metadata
-     */
-    public function __construct($metadata)
-    {
-        $this->metadata = $metadata;
+    public function __construct(
+        private ClassMetadata|LegacyClassMetadata $metadata
```

Comment:
Note to self: we may be able to remove the `LegacyClassMetedata` - depends on doctrine version constraints...